synapseclient/extensions/curator/schema_generation.py

```diff
@@ -4790,6 +4854,30 @@ def __post_init__(self) -> None:
             explicit_maximum if explicit_maximum is not None else implicit_maximum
         )
 
+        if column_pattern and column_type and column_type.value != "string":
```

Comment:
Could you move this into `self._validate_column_type_compatibility`, see line 4814 above where it's called.